service/src/main/java/uk/nhs/adaptors/gp2gp/transformJsonToXmlTool/TransformJsonToXml.java

```diff
@@ -0,0 +1,237 @@
+                new BloodPressureMapper(
+                        messageContext, randomIdGeneratorService, new StructuredObservationValueMapper(),
+                        codeableConceptCdMapper, new ParticipantMapper()),
+                new ConditionLinkSetMapper(
+                        messageContext, randomIdGeneratorService, codeableConceptCdMapper, participantMapper),
+                new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new DocumentReferenceToNarrativeStatementMapper(messageContext, supportedContentTypes),
+                new ImmunizationObservationStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new MedicationStatementMapper(messageContext, codeableConceptCdMapper, participantMapper, randomIdGeneratorService),
+                new ObservationToNarrativeStatementMapper(messageContext, participantMapper),
+                new ObservationStatementMapper(
+                        messageContext,
+                        new StructuredObservationValueMapper(),
+                        new PertinentInformationObservationValueMapper(),
+                        codeableConceptCdMapper,
+                        participantMapper
+                ),
+                new RequestStatementMapper(messageContext, codeableConceptCdMapper, participantMapper),
+                new DiagnosticReportMapper(messageContext, specimenMapper, participantMapper, randomIdGeneratorService)
+        );
+
+        final EncounterMapper encounterMapper = new EncounterMapper(messageContext, encounterComponentsMapper);
+
+        final NonConsultationResourceMapper nonConsultationResourceMapper =
+                new NonConsultationResourceMapper(messageContext, randomIdGeneratorService, encounterComponentsMapper);
+
+        final AgentPersonMapper agentPersonMapper = new AgentPersonMapper(messageContext);
+
+        final AgentDirectoryMapper agentDirectoryMapper = new AgentDirectoryMapper(messageContext,
+                agentPersonMapper);
+
+        EhrExtractMapper ehrExtractMapper = new EhrExtractMapper(randomIdGeneratorService, timestampService, encounterMapper,
+                nonConsultationResourceMapper, agentDirectoryMapper, messageContext);
+
+        final EhrExtractTemplateParameters ehrExtractTemplateParameters =
+                ehrExtractMapper.mapBundleToEhrFhirExtractParams(getGpcStructuredTaskDefinition, bundle);
+
+        final String ehrExtractContent = ehrExtractMapper.mapEhrExtractToXml(ehrExtractTemplateParameters);
+
+        final String hl7TranslatedResponse = outputMessageWrapperMapper.map(getGpcStructuredTaskDefinition, ehrExtractContent);
+
+        messageContext.resetMessageContext();
+
+        return hl7TranslatedResponse;
+    }
+
+    @Data
+    @Builder
+    @AllArgsConstructor
+    public static class InputWrapper {
+        private List<String> jsonFileNames;
```

Comment:
if InputWrapper would have only 2 strings, jsonInputFile and jsonFileName it would simplify the code much.

So in general you would have 
```
public static class InputFile {
private String jsonFileName;
 private String jsonFileInput;
}
```

then method getFile() would return list of InputFile

```
 private static List<InputFile> getFiles() {
 
 ...
 
 
 return  Arrays.stream(files)
                .peek(file -> LOGGER.info("Parsing file: {}", file.getName()))
                .filter(file -> FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("json"))
                .map(file -> {
                    try {
                        String jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + file.getName());
                        return new InputFile(file.getName(), jsonAsString);
                    } catch (IOException e) {
                       /// throw exception
                    }
                   
                }).collect(Collectors.toList());

```

then in main only:

```
if (startTest.equals("True")) {
                getFiles().forEach(file -> {
                    String xmlResult = mapJsonToXml(file.getJsonFileInput());
                    writeToFile(xmlResult, file.getJsonFileName());
                });
            }
 ```